cli/app/commands/conflict/conflict.py

```diff
@@ -0,0 +1,391 @@
+import os
+import subprocess
+import re
+import yaml
+from typing import Dict, List, Optional, Any, Tuple
+from packaging import version
+from packaging.specifiers import SpecifierSet
+from packaging.version import Version
+
+from app.utils.logger import Logger
+from app.utils.protocols import LoggerProtocol
+from app.utils.output_formatter import OutputFormatter
+from app.utils.lib import ParallelProcessor
+from .models import ConflictCheckResult, ConflictConfig
+from .messages import (
+    error_checking_tool_version,
+    error_parsing_version,
+    timeout_checking_tool,
+    tool_not_found,
+    tool_version_mismatch,
+    tool_version_compatible,
+    conflict_config_not_found,
+    conflict_invalid_config,
+    conflict_loading_config,
+    conflict_config_loaded,
+)
+
+
+class VersionParser:
+    """Utility class for parsing and comparing versions."""
+    
+    # Version pattern mappings for different tools
+    VERSION_PATTERNS = [
+        r"version\s+(\d+\.\d+\.\d+)",
+        r"v(\d+\.\d+\.\d+)",
+        r"(\d+\.\d+\.\d+)",
+        r"Version\s+(\d+\.\d+\.\d+)",
+        r"(\d+\.\d+)",
+    ]
+    
+    @staticmethod
+    def parse_version_output(tool: str, output: str) -> Optional[str]:
+        """Parse version from tool output."""
+        try:
+            # Common version patterns
+            for pattern in VersionParser.VERSION_PATTERNS:
+                match = re.search(pattern, output, re.IGNORECASE)
+                if match:
+                    return match.group(1)
+
+            # Tool-specific parsing
+            if tool == "go":
+                match = re.search(r"go(\d+\.\d+\.\d+)", output)
+                if match:
+                    return match.group(1)
+
+            return None
+        except Exception as e:
+            raise ValueError(f"Error parsing version for {tool}: {str(e)}")
+
+    @staticmethod
+    def compare_versions(current: str, expected: str) -> bool:
+        """Compare version against requirement specification."""
+        try:
+            # Handle simple version comparisons (backwards compatibility)
+            if not any(op in expected for op in [">=", "<=", ">", "<", "==", "!=", "~", "^"]):
+                # Default to >= for simple version strings
+                return version.parse(current) >= version.parse(expected)
+
+            # Handle version ranges and specifiers
+            spec_set = SpecifierSet(expected)
+            return Version(current) in spec_set
+
+        except Exception:
+            # Fallback to string comparison
+            return current == expected
+
+    @staticmethod
+    def normalize_version_requirement(requirement: str) -> str:
+        """Parse version requirement and return a normalized specifier."""
+        if not requirement:
+            return requirement
+
+        # Handle npm-style caret ranges: ^1.20.0 -> >=1.20.0, <2.0.0
+        if requirement.startswith("^"):
```

Comment:
i can see the same set of code is being repeated from line 95 to 103, and again in 111 to 120, which can be simplified and abstracted into single function, no repeatation